src/ibex_bluesky_core/callbacks/fitting/livefit_logger.py

```diff
@@ -0,0 +1,141 @@
+    """Generates files as part of a scan that describe the fit(s) which have been performed."""
+
+    def __init__(
+        self,
+        livefit: LiveFit,
+        y: str,
+        x: str,
+        output_dir: Path,
+        postfix: str,
+        yerr: str | None = None,
+    ) -> None:
+        """Initialise LiveFitLogger callback.
+
+        Args:
+            livefit (LiveFit): A reference to LiveFit callback to collect fit info from.
+            y (str): The name of the signal pointing to y counts data.
+            x (str): The name of the signal pointing to x counts data.
+            output_dir (str): A path to where the fitting file should be stored.
+            postfix (str): A small string that should be placed at the end of the
+                filename to disambiguate multiple fits and avoid overwriting.
+            yerr (str): The name of the signal pointing to y count uncertainties data.
+
+        """
+        super().__init__()
+        self.livefit = livefit
+        self.postfix = postfix
+        self.output_dir = output_dir
+        self.current_start_document: Optional[str] = None
+
+        self.x = x
+        self.y = y
+        self.yerr = yerr
+
+        self.x_data = []
+        self.y_data = []
+        self.yerr_data = []
+
+    def start(self, doc: RunStart) -> None:
+        """Create the output directory if it doesn't already exist then setting the filename.
+
+        Args:
+            doc (RunStart): The start bluesky document.
+
+        """
+        self.output_dir.mkdir(parents=True, exist_ok=True)
+        self.current_start_document = doc[UID]
+        self.filename = self.output_dir / f"{self.current_start_document}{self.postfix}.csv"
+
+    def event(self, doc: Event) -> Event:
+        """Start collecting, y, x and yerr data.
+
+        Args:
+            doc: (Event): An event document.
+
+        """
+        event_data = doc[DATA]
+
+        assert self.x in event_data, f"{self.x} is not in event document."
+        assert self.y in event_data, f"{self.y} is not in event document."
+
+        self.x_data.append(event_data[self.x])
+        self.y_data.append(event_data[self.y])
+
+        if self.yerr is not None:
+            assert self.yerr in event_data, f"{self.yerr} is not in event document."
+            self.yerr_data.append(event_data[self.yerr])
+
+        return super().event(doc)
+
+    def stop(self, doc: RunStop) -> None:
+        """Write to the fitting file.
+
+        Args:
+            doc (RunStop): The stop bluesky document.
+
+        """
+        if self.livefit.result is None:
+            logger.error("LiveFit.result was None. Could not write to file.")
+            return
+
+        # Evaluate the model function for each x point
+        kwargs = {"x": np.array(self.x_data)}
+        kwargs.update(self.livefit.result.values)
+        self.y_fit_data = self.livefit.result.model.eval(**kwargs)
+
+        self.stats = str(self.livefit.result.fit_report()).split("\n")
```

Comment:
Not sure why, but correlations are formatted a bit funny here:

```
...
[[Variables]]
    amp:        -2.7062e-10 +/- 2.3997e-11 (8.87%) (init = 100)
    sigma:       4.54685385 +/- 0.47742330 (10.50%) (init = 3.191714)
    x0:          4.47700894 +/- 0.52513584 (11.73%) (init = 5)
    background:  50.0000000 +/- 2.2286e-11 (0.00%) (init = 50)
[[Correlations]] (unreported correlations are < 0.100)
"    C(amp, x0)           = -0.6614"
"    C(amp, background)   = -0.3537"
"    C(sigma, background) = +0.3526"
"    C(amp, sigma)        = -0.1247"

...
```

i.e. leading/trailing quotes?